From c530559318bcc4bf4d77321793be4f7e84c00ae9 Mon Sep 17 00:00:00 2001
From: "Lu,Yang A" <yang.a.lu@intel.com>
Date: Tue, 12 May 2020 02:24:54 +0800
Subject: [PATCH] Modify edid to 1080P 62.5Hz

With QEMU edid disabled, modify host edid to 1080P to set
GVT-g Android resolution to 1080P. Set fresh-rate to 62.5Hz
to align with GVT-g's vsync setting.

Tracked-On: OAM-90886
Signed-off-by: Lu Yang A <yang.a.lu@intel.com>
---
 drivers/gpu/drm/i915/gvt/display.c | 46 ++++++++------------------------------
 1 file changed, 9 insertions(+), 37 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/display.c b/drivers/gpu/drm/i915/gvt/display.c
index a62bdf9..4edeb5f 100644
--- a/drivers/gpu/drm/i915/gvt/display.c
+++ b/drivers/gpu/drm/i915/gvt/display.c
@@ -119,43 +119,15 @@ static unsigned char virtual_dp_monitor_edid[GVT_EDID_NUM][EDID_SIZE] = {
 		0xef,
 	},
 	{
-/* EDID with 1920x1200 as its resolution */
-		/*Header*/
-		0x00, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00,
-		/* Vendor & Product Identification */
-		0x22, 0xf0, 0x54, 0x29, 0x00, 0x00, 0x00, 0x00, 0x04, 0x17,
-		/* Version & Revision */
-		0x01, 0x04,
-		/* Basic Display Parameters & Features */
-		0xa5, 0x34, 0x20, 0x78, 0x23,
-		/* Color Characteristics */
-		0xfc, 0x81, 0xa4, 0x55, 0x4d, 0x9d, 0x25, 0x12, 0x50, 0x54,
-		/* Established Timings: maximum resolution is 1024x768 */
-		0x21, 0x08, 0x00,
-		/*
-		 * Standard Timings.
-		 * below new resolutions can be supported:
-		 * 1920x1080, 1280x720, 1280x960, 1280x1024,
-		 * 1440x900, 1600x1200, 1680x1050
-		 */
-		0xd1, 0xc0, 0x81, 0xc0, 0x81, 0x40, 0x81, 0x80, 0x95, 0x00,
-		0xa9, 0x40, 0xb3, 0x00, 0x01, 0x01,
-		/* 18 Byte Data Blocks 1: max resolution is 1920x1200 */
-		0x28, 0x3c, 0x80, 0xa0, 0x70, 0xb0,
-		0x23, 0x40, 0x30, 0x20, 0x36, 0x00, 0x06, 0x44, 0x21, 0x00, 0x00, 0x1a,
-		/* 18 Byte Data Blocks 2: invalid */
-		0x00, 0x00, 0x00, 0xfd, 0x00, 0x18, 0x3c, 0x18, 0x50, 0x11, 0x00, 0x0a,
-		0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-		/* 18 Byte Data Blocks 3: invalid */
-		0x00, 0x00, 0x00, 0xfc, 0x00, 0x48,
-		0x50, 0x20, 0x5a, 0x52, 0x32, 0x34, 0x34, 0x30, 0x77, 0x0a, 0x20, 0x20,
-		/* 18 Byte Data Blocks 4: invalid */
-		0x00, 0x00, 0x00, 0xff, 0x00, 0x43, 0x4e, 0x34, 0x33, 0x30, 0x34, 0x30,
-		0x44, 0x58, 0x51, 0x0a, 0x20, 0x20,
-		/* Extension Block Count */
-		0x00,
-		/* Checksum */
-		0x45,
+/* EDID with 1920x1080 as its resolution */
+		0x00, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x1e, 0xd4, 0x88, 0x88, 0x00, 0x00, 0x00, 0x00,
+		0x0f, 0x1e, 0x01, 0x04, 0xa5, 0x34, 0x20, 0x78, 0x23, 0xfc, 0x85, 0xa4, 0x55, 0x4d, 0x9d, 0x25,
+		0x12, 0x50, 0x54, 0x00, 0x00, 0x00, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+		0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x72, 0x38, 0x80, 0xa0, 0x70, 0x38, 0x20, 0x40, 0x30, 0x20,
+		0x35, 0x00, 0x40, 0x44, 0x21, 0x00, 0x00, 0x1a, 0x00, 0x00, 0x00, 0xfd, 0x00, 0x18, 0x3c, 0x18,
+		0x50, 0x11, 0x00, 0x0a, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x00, 0x00, 0x00, 0xfc, 0x00, 0x47,
+		0x56, 0x54, 0x20, 0x56, 0x69, 0x72, 0x74, 0x0a, 0x20, 0x20, 0x20, 0x20, 0x00, 0x00, 0x00, 0xff,
+		0x00, 0x4b, 0x56, 0x4d, 0x47, 0x56, 0x54, 0x31, 0x32, 0x33, 0x34, 0x35, 0x0a, 0x20, 0x00, 0x02,
 	},
 };
 
-- 
2.7.4

